app/Http/Controllers/UserController.php

```diff
@@ -47,8 +51,32 @@ public function followUser(User $profileId)
 
     public function unFollowUser(User $profileId)
     {
-        
         $profileId->followers()->detach(auth()->user()->id);
         return redirect()->back()->with('success', 'Successfully unfollowed the user.');
     }
+
+    public function showCategories()
+    {
+        $categories = Category::all();
+
+        return view('users.Lesson',compact('categories'));
+    }
+
+    public function makeLesson($categoryId)
+    {
+        $user = auth()->user();
+    
+        $lesson =Lesson::create([
+            'score' => 0,
+            'category_id' => $categoryId,
+            'user_id' =>$user->id,
+        ]);
+       
+        $lessonId=$lesson->id;
+        // dd($questions);
+        // $choices = $questions->getChoices()->get();
```

Comment:
always check if you have commented code. Not just that always check if your code is clean so that we will refrain from redo